File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/java/F09.P1.java
class Main {
    static boolean probe() {
        String s1 = new StringBuilder("ab").append("c").toString();
        String s2 = new StringBuilder("ab").append("c").toString();
        // BEGIN PROBE F09.P1
        boolean eq = s1.equals(s2);
        // END PROBE F09.P1
        return eq;
    }

    public static void main(String[] args) {
        System.out.println(probe());
    }
}

File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/kotlin/F09.P1.kt
fun probe(): Boolean {
    val s1 = StringBuilder("ab").append("c").toString()
    val s2 = StringBuilder("ab").append("c").toString()
    // BEGIN PROBE F09.P1
    val eq = s1 == s2
    // END PROBE F09.P1
    return eq
}

fun main() {
    println(probe())
}
